src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java

```diff
@@ -335,10 +338,32 @@ public ResponseEntity<GenerateAssetResponseDTO> generateAsset(
             throw new NotFoundException("Source Container: " + generateAssetDTO.getSourceContainer());
         }
 
+        // 404s of there are no mp4 files in the container
+        // this is called in the MK process anyway but AMS doesn't need the specific filename
+        azureFinalStorageService.getMp4FileName(generateAssetDTO.getSourceContainer());
+
         log.info("Attempting to generate asset: {}", generateAssetDTO);
 
         var result = mediaServiceBroker.getEnabledMediaService().importAsset(generateAssetDTO);
         if (result.getJobStatus().equals(JobState.FINISHED.toString())) {
+            // add new version to recording etc
+            var parentRecording = recordingService.findById(UUID.fromString(generateAssetDTO.getSourceContainer()));
+
+            var recording = new CreateRecordingDTO();
+            recording.setId(UUID.fromString(generateAssetDTO.getDestinationContainer()));
+            recording.setParentRecordingId(parentRecording.getId());
+            recording.setCaptureSessionId(parentRecording.getCaptureSession().getId());
+
+            var search = new SearchRecordings();
+            search.setCaptureSessionId(parentRecording.getCaptureSession().getId());
+            var numRecordingsForCaptureSession = recordingService.findAll(
+                search,
+                false,
```

Comment:
We should include the delete recordings since they still count as edits and will mess up the version numbering.
I.e. if you have a recording with 5 versions (v1-5) and delete v4. This search will currently return a list of size 4 meaning the next version number will be 5. The problem is we already have a v5 recording so the count is off. 
```suggestion
                true,
```